### PR TITLE
sdrangel: fix darwin build

### PR DESCRIPTION
**Pull request by mrene**
###### Description of changes
This makes a few fixes in order to get sdrangel to build and run correction on aarch64-darwin.

1. QtWebEngine has had a fix in #118084 in order to fix version mismatches in Qt, but this fix was introduced inside a conditional block only applying it on linux. This makes the fix available to other platforms as well, while not causing any rebuilds for linux platforms (by preserving the whitespace content of the linux derivation inputs)
2. OpenCV3 was building without Video IO on darwin. OpenCV4 had the right configuration, but not v3. I added the right dependencies and it now builds correctly.
3. A few dependencies were marked as `platforms = platforms.linux`, since they build correctly on darwin, I upgraded their platform metadata to `platforms.unix`

I have tested applying #220746 as well (another sdrangel fix PR), and it builds correctly in both platforms.

- Built on platform(s)
  - [x] x86_64-linux (no changes, same derivation)
  - [x] aarch64-linux (no changes, same derivation) 
  - [ ] x86_64-darwin (qtwebengine is marked as broken, and doesn't build)
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? No
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/radio/sdrangel/default.nix [hunk 1]
```diff
@@ -95,7 +96,7 @@ mkDerivation rec {
     "-DDAB_LIB=${dab_lib}"
     "-DLIBSERIALDV_INCLUDE_DIR:PATH=${serialdv}/include/serialdv"
     "-DLIMESUITE_INCLUDE_DIR:PATH=${limesuite}/include"
-    "-DLIMESUITE_LIBRARY:FILEPATH=${limesuite}/lib/libLimeSuite.so"
+    ("-DLIMESUITE_LIBRARY:FILEPATH=${limesuite}/lib/libLimeSuite." + (if stdenv.isDarwin then "dylib" else "so"))
```
Comment (wegank) [inline pkgs/applications/radio/sdrangel/default.nix:99]:
```suggestion
    "-DLIMESUITE_LIBRARY:FILEPATH=${limesuite}/lib/libLimeSuite${stdenv.hostPlatform.extensions.sharedLibrary}"
```

Comment (mrene) [reply pkgs/applications/radio/sdrangel/default.nix]: That's where the helper was! Fixed!

**Comment by wegank** [comment]
@ofborg eval

**Review by wegank — APPROVED** [review]

**Review by alkeryn — APPROVED** [review]
haven't tested it but code looks good !

**Comment by mrene** [comment]
Result of `nixpkgs-review pr 220893` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>29 packages marked as broken and skipped:</summary>
  <ul>
    <li>caffeWithCuda</li>
    <li>caffeWithCuda.bin</li>
    <li>gimp-with-plugins</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>libsForQt5.alkimia</li>
    <li>libsForQt5.falkon</li>
    <li>libsForQt5.tokodon</li>
    <li>lightly-qt</li>
    <li>patray</li>
    <li>patray.dist</li>
    <li>plasma5Packages.alkimia</li>
    <li>plasma5Packages.falkon</li>
    <li>plasma5Packages.tokodon</li>
    <li>python310Packages.bpycv</li>
    <li>python310Packages.bpycv.dist</li>
    <li>python310Packages.caffe</li>
    <li>python310Packages.caffe.bin</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.mask-rcnn</li>
    <li>python310Packages.mask-rcnn.dist</li>
    <li>python310Packages.pywebview</li>
    <li>python310Packages.pywebview.dist</li>
    <li>python311Packages.pywebview</li>
    <li>python311Packages.pywebview.dist</li>
    <li>streamdeck-ui</li>
    <li>streamdeck-ui.dist</li>
    <li>subsurface</li>
  </ul>
</details>
<details>
  <summary>24 packages failed to build:</summary>
  <ul>
    <li>PageEdit</li>
    <li>frescobaldi</li>
    <li>frescobaldi.dist</li>
    <li>mellowplayer</li>
    <li>merkaartor</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>onionshare-gui</li>
    <li>onionshare-gui.dist</li>
    <li>python-qt</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.magicgui.dist</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.napari-npe2.dist</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.stytra</li>
    <li>python310Packages.stytra.dist</li>
    <li>qsyncthingtray</li>
    <li>qutebrowser</li>
    <li>qutebrowser.dist</li>
    <li>sl1-to-photon</li>
    <li>tumpa</li>
    <li>tumpa.dist</li>
  </ul>
</details>
<details>
  <summary>67 packages built:</summary>
  <ul>
    <li>anki</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>aptdec</li>
    <li>caffe</li>
    <li>caffe.bin</li>
    <li>cm256cc</li>
    <li>dab_lib</li>
    <li>dsdcc</li>
    <li>easyocr (python310Packages.easyocr)</li>
    <li>easyocr.dist (python310Packages.easyocr.dist)</li>
    <li>ghostwriter</li>
    <li>gimpPlugins.gmic</li>
    <li>gmic-qt</li>
    <li>hydrus</li>
    <li>hydrus.doc</li>
    <li>qt5.qtspeech (libsForQt5.qtspeech)</li>
    <li>qt5.qtspeech.dev (libsForQt5.qtspeech.dev)</li>
    <li>qt5.qtwebengine (libsForQt5.qtwebengine)</li>
    <li>qt5.qtwebengine.bin (libsForQt5.qtwebengine.bin)</li>
    <li>qt5.qtwebengine.dev (libsForQt5.qtwebengine.dev)</li>
    <li>qt5.qtwebview (libsForQt5.qtwebview)</li>
    <li>qt5.qtwebview.bin (libsForQt5.qtwebview.bin)</li>
    <li>qt5.qtwebview.dev (libsForQt5.qtwebview.dev)</li>
    <li>mindforger</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>mxnet</li>
    <li>opencv3 (opencv3WithoutCuda)</li>
    <li>openshot-qt</li>
    <li>openshot-qt.dist</li>
    <li>python310Packages.dremel3dpy</li>
    <li>python310Packages.dremel3dpy.dist</li>
    <li>python310Packages.imagecorruptions</li>
    <li>python310Packages.imagecorruptions.dist</li>
    <li>python310Packages.imantics</li>
    <li>python310Packages.imantics.dist</li>
    <li>python310Packages.imutils</li>
    <li>python310Packages.imutils.dist</li>
    <li>python310Packages.mxnet</li>
    <li>python310Packages.mxnet.dist</li>
    <li>python310Packages.opencv3</li>
    <li>python310Packages.pyqtwebengine</li>
    <li>python310Packages.pyqtwebengine.dev</li>
    <li>python310Packages.pyqtwebengine.dist</li>
    <li>python310Packages.vidstab</li>
    <li>python310Packages.vidstab.dist</li>
    <li>python311Packages.imantics</li>
    <li>python311Packages.imantics.dist</li>
    <li>python311Packages.imutils</li>
    <li>python311Packages.imutils.dist</li>
    <li>python311Packages.mxnet</li>
    <li>python311Packages.mxnet.dist</li>
    <li>python311Packages.opencv3</li>
    <li>python311Packages.pyqtwebengine</li>
    <li>python311Packages.pyqtwebengine.dev</li>
    <li>python311Packages.pyqtwebengine.dist</li>
    <li>python311Packages.vidstab</li>
    <li>python311Packages.vidstab.dist</li>
    <li>sdrangel</li>
    <li>serialdv</li>
    <li>sgp4</li>
    <li>vivisect</li>
    <li>vivisect.dist</li>
    <li>waifu2x-converter-cpp</li>
    <li>webmacs</li>
    <li>webmacs.dist</li>
  </ul>
</details>

**Comment by mrene** [comment]
I did a check check and the 24 package build failiures above are also all broken on master (most due to broken transitive deps on aarch64-darwin).

**Pull request merged**